.github/workflows/node.js.yml

```diff
@@ -10,6 +10,42 @@ on:
         branches: [master, feature/*, mynah-dev]
 
 jobs:
+    check-test-files:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4
+              with:
+                  fetch-depth: 2
+            - name: Check for the presence test files
+              id: check_file
+              run: |
+                  # Check if there are any changes in the src folder
+                  sourceFiles=$(git diff --name-only ${{ github.event.pull_request.base.sha }} | { grep 'src' || true; })
+                  if [ -z "$sourceFiles" ]; then
+                      echo "Did not find src files in the code changes"
+                      exit 0
+                  fi
+
+                  # Of those changes in the src folder, check if any of them are in *.test.ts files
+                  testFiles=$(echo $sourceFiles | { grep '\.test\.ts$' || true; })
+                  if [ -n "$testFiles" ]; then
+                      echo "Found test files in the code changes"
+                      exit 0
+                  fi
+                  echo "TEST_FILES_FOUND=true" >> $GITHUB_OUTPUT
+            - name: Notify contributor
+              if: steps.check_file.outputs.TEST_FILES_FOUND == 'true'
+              uses: actions/github-script@v7
+              with:
+                  github-token: ${{ secrets.GITHUB_TOKEN }}
+                  script: |
+                      github.rest.issues.createComment({
+                          issue_number: context.issue.number,
+                          owner: context.repo.owner,
+                          repo: context.repo.repo,
+                          body: 'We have detected that your changes do not add or modify test files. Please add or update tests, or indicate why tests are not needed.'
```

Comment:
Now that conventional commits are checked https://github.com/aws/aws-toolkit-vscode/pull/5458 , we could also leave a note requesting a changelog item for `feat` and `fix` PRs.

Reply:
Should we do that separately inside of the lintcommit file? It would actually be reasonably easy to get that working

Reply:
Not sure. We can try either way and see what works best :)